drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -417,28 +417,23 @@ int lis2dh_init(const struct device *dev)
 	})
 
 #ifdef CONFIG_LIS2DH_TRIGGER
-#define LIS2DH_HAS_IRQ_IDX(inst, idx)					\
-		DT_INST_PROP_HAS_IDX(inst, irq_gpios, idx)
+#define LIS2DH_SET_TRIG_BY_IDX(id, prop, idx)				\
+	COND_CODE_1(DT_INST_PROP_HAS_IDX(id, prop, idx),		\
+		    (true),						\
+		    (false))
```

Comment:
Yes, of course. There is no need for LIS2DH_SET_TRIG_BY_IDX at all, I can simply use DT_INST_PROP_HAS_IDX. The code was initially different and when I simplified it I came up with this stupid stuff. Yesterday was a bad day for me... sorry.

Reply:
Thanks for pointing out!

Reply:
> This could be simplified to just `DT_INST_PROP_HAS_IDX(id, prop, idx)`. I don't think you need the `COND_CODE_1`

In latest submitted code I took care also for this suggestions. Thanks!